AppCenter/AppCenter/Internals/Util/MSJwtClaims.m

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import "MSJwtClaims.h"
+#import "MSAppCenterInternal.h"
+#import "MSLogger.h"
+#import <Foundation/Foundation.h>
+
+static NSString *const kMSJwtPartsSeparator = @".";
+static NSString *const kMSSubjectClaim = @"sub";
+static NSString *const kMSExpirationClaim = @"exp";
+
+@implementation MSJwtClaims
+
+- (instancetype)initWithSubject:(NSString *)subject expirationDate:(NSDate *)expirationDate {
+  self = [super init];
+  if (self) {
+    _subject = subject;
+    _expirationDate = expirationDate;
+  }
+  return self;
+}
+
++ (MSJwtClaims *)parse:(NSString *)jwt {
+  NSArray *parts = [jwt componentsSeparatedByString:kMSJwtPartsSeparator];
+  if ([parts count] < 2) {
+    MSLogError([MSAppCenter logTag], @"Failed to parse JWT, not enough parts.");
+    return nil;
+  }
+  NSString *base64ClaimsPart = parts[1];
+  @try {
+    NSError *error;
+    NSData *claimsPartData = [[NSData alloc] initWithBase64EncodedString:base64ClaimsPart options:0];
+    NSDictionary *claims = [NSJSONSerialization JSONObjectWithData:claimsPartData options:0 error:&error];
+    if ([claims objectForKey:kMSExpirationClaim] == nil || [claims objectForKey:kMSSubjectClaim] == nil) {
+      MSLogError([MSAppCenter logTag], @"Deserialized claims is missing `sub` or `exp`");
+      return nil;
+    }
+
+    // We will treat invalid expiration times as an interval of 0 because there is not an easy way to determine the
```

Comment:
What does it mean by `an interval of 0`?

Reply:
Interval of 0 since the epoch (1970)